rskj-core/src/main/java/org/ethereum/rpc/JsonRpcWeb3ServerHandler.java

```diff
@@ -0,0 +1,123 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger("jsonrpc");
+    private static final int JSON_RPC_PARSE_ERROR_CODE = -32700;
+    private static final int JSON_RPC_INVALID_PARAMS_CODE = -32602;
+    private static final int JSON_RPC_INTERNAL_ERROR_CODE = -32603;
+    private static final int JSON_RPC_SERVER_ERROR_LOW_CODE = -32000;
+    private static final int JSON_RPC_SERVER_ERROR_HIGH_CODE = -32099;
+    private static final int JSON_RPC_INVALID_REQUEST_CODE = -32600;
+    private static final int JSON_RPC_METHOD_NOT_FOUND_CODE = -32601;
+
+    private final ObjectMapper mapper = new ObjectMapper();
+    private final JsonNodeFactory jsonNodeFactory = JsonNodeFactory.instance;
+    private final JsonRpcFilterServer jsonRpcServer;
+    private final CorsConfiguration corsConfiguration;
+
+    public JsonRpcWeb3ServerHandler(Web3 service, List<ModuleDescription> filteredModules, CorsConfiguration corsConfiguration) {
+        this.corsConfiguration = corsConfiguration;
+        this.jsonRpcServer = new JsonRpcFilterServer(service, service.getClass(), filteredModules);
+        jsonRpcServer.setErrorResolver(new MultipleErrorResolver(new RskErrorResolver(), AnnotationsErrorResolver.INSTANCE, DefaultErrorResolver.INSTANCE));
+    }
+
+    @Override
+    protected void channelRead0(ChannelHandlerContext ctx, FullHttpRequest request) throws Exception {
+        HttpMethod httpMethod = request.getMethod();
+        HttpResponse response;
+        if (HttpMethod.OPTIONS.equals(httpMethod)) {
+            response = new DefaultFullHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.OK);
+            response.headers()
+                .add(HttpHeaders.Names.ACCESS_CONTROL_ALLOW_METHODS, String.join(", ", HttpMethod.POST.name(), HttpMethod.GET.name(), HttpMethod.OPTIONS.name()))
+                .add(HttpHeaders.Names.ACCESS_CONTROL_ALLOW_HEADERS, HttpHeaders.Names.CONTENT_TYPE)
+                .add(HttpHeaders.Names.ACCESS_CONTROL_MAX_AGE, String.valueOf(600))
+                .add(HttpHeaders.Names.ALLOW, String.join(", ", HttpMethod.GET.name(), HttpMethod.HEAD.name(), HttpMethod.POST.name(), HttpMethod.TRACE.name(), HttpMethod.OPTIONS.name()));
+        } else if (HttpMethod.POST.equals(httpMethod)) {
+            ByteBuf body = request.content();
+            byte[] rawBody = new byte[body.readableBytes()];
+            int readerIndex = body.readerIndex();
+            body.getBytes(readerIndex, rawBody);
+
+            ByteBuf content = Unpooled.buffer();
+            HttpResponseStatus responseStatus = HttpResponseStatus.OK;
+            try (ByteBufOutputStream os = new ByteBufOutputStream(content)){
+                JsonNode jsonRpcRequest = mapper.readTree(rawBody);
+                int result = jsonRpcServer.handleNode(jsonRpcRequest, os);
+                if (result != 0) {
+                    if (result == JSON_RPC_PARSE_ERROR_CODE ||
```

Comment:
Maybe, Instead of doing this you could model the response codes with an Enum, and each response knows to which http status gets mapped, so you could get something like 

responseStatus = SarasaEnum.velueOf(result).httpStatus()

Or something like that...